servant-jquery/test/Servant/JQuerySpec.hs

```diff
@@ -46,51 +49,107 @@ customHeaderProxy = Proxy
 customHeaderProxy2 :: Proxy CustomHeaderAPI2
 customHeaderProxy2 = Proxy
 
+data TestNames = Vanilla
+               | VanillaCustom
+               | JQuery
+               | JQueryCustom
+               | Angular
+               | AngularCustom
+                 deriving (Show, Eq)
+
+customOptions :: CommonGeneratorOptions
+customOptions = defCommonGeneratorOptions {
+            successCallback = "okCallback",
+            errorCallback = "errorCallback"
+        }
+                 
 spec :: Spec
-spec = describe "Servant.JQuery"
-    generateJSSpec
-
-generateJSSpec :: Spec
-generateJSSpec = describe "generateJS" $ do
+spec = describe "Servant.JQuery" $ do
+    (generateJSSpec Vanilla JS.generateVanillaJS)
+    (generateJSSpec VanillaCustom $ JS.generateVanillaJSWith customOptions)
+    (generateJSSpec JQuery JQ.generateJQueryJS)
+    (generateJSSpec JQueryCustom $ JQ.generateJQueryJSWith customOptions)
+    (generateJSSpec Angular $ NG.generateAngularJS NG.defAngularOptions)
+    (generateJSSpec AngularCustom $ (NG.generateAngularJSWith NG.defAngularOptions) customOptions)
+    
+    (angularSpec Angular)
+    (angularSpec AngularCustom)
+
+angularSpec :: TestNames -> Spec    
+angularSpec test = describe specLabel $ do
+    it "should implement a service globally" $ do
+        let jsText = genJS $ listFromAPI (Proxy :: Proxy TestAPI)
+        output jsText
+        jsText `shouldContain` (".service('" ++ testName ++ "'")
+        
+    it "should depend on $http service globally" $ do
+        let jsText = genJS $ listFromAPI (Proxy :: Proxy TestAPI)
+        output jsText
+        jsText `shouldContain` ("('" ++ testName ++ "', function($http) {")
+        
+    it "should not depend on $http service in handlers" $ do
+        let jsText = genJS $ listFromAPI (Proxy :: Proxy TestAPI)
+        output jsText
+        jsText `shouldNotContain` "getsomething($http, "
+    where
+        specLabel = "generateJS(" ++ (show test) ++ ")"
+        --output = putStrLn
+        output _ = return ()
+        testName = "MyService"
+        ngOpts = NG.defAngularOptions { NG.serviceName = testName }
+        genJS req = NG.wrapInService ngOpts req
+    
+generateJSSpec :: TestNames -> (AjaxReq -> String) -> Spec
+generateJSSpec n gen = describe specLabel $ do
```

Comment:
Can we tweak the tests so that if one fails with a specific `TestName` constructor, we get to see what the constructor is?

(Possibly this could be done by wrapping all the `it`s, in a `context (show name)` do block.